client/src/services/DonuService.ts

```diff
@@ -12,19 +13,85 @@ const callDonu = (request: Donu.Request): Promise<Donu.Response> => {
   }
 };
 
+export const getDonuModelSource = async (model: string, type: Donu.Type): Promise<Donu.ModelGraph> => {
+  const request: Donu.Request = {
+    command: Donu.RequestCommand.GET_MODEL_SOURCE,
+    definition: {
+      type,
+      source: {
+        model,
+      },
+    },
+  };
+  const response = await callDonu(request);
+  if (response.status === Donu.ResponseStatus.success) {
+    const modelSource = response.result.source;
+    return JSON.parse(modelSource);
+  } else {
+    console.error('[DONU Service] — fetchDonuModelSource', response);
+  }
+};
+
+const populateDonuModelWithGrometSource = async (model: Donu.ModelDefinition): Promise<void> => {
+  const gromet = await getDonuModelSource(model.source.model, model.type);
+  (model as any).gromet = gromet;
+};
+
 /** Fetch a complete list of available models from Donu API */
-export const fetchDonuModels = async (): Promise<any[]> => {
+export const fetchDonuModels = async (): Promise<Model.Model[]> => {
   const request: Donu.Request = {
     command: Donu.RequestCommand.LIST_MODELS,
   };
 
   const response = await callDonu(request);
   if (response.status === Donu.ResponseStatus.success) {
-    const models = response?.result as Donu.ModelDefinition[] ?? null;
-    const grometOnlyModels = models.filter(model => {
-      return [Donu.Type.GROMET_PNC/*, Donu.Type.GROMET_FN */].includes(model.type);
+    // 1. Get models
+    let models = response?.result as any ?? null;
```

Comment:
You can use `Donu.ResponseResult` instead of `any` for the type of the Donu response result.

Reply:
Why removing the type instead of using `Donu.ResponseResult`?